lib/modules/versioning/distro.ts

```diff
@@ -58,7 +100,59 @@ export class DistroInfo {
     return input;
   }
 
+  /**
+   * Get schedule of a given release
+   * @param input A codename/semVer
+   * @returns A schedule if available, otherwise undefined
+   */
   public getSchedule(input: string): DistroSchedule {
-    return this._distroInfo[input];
+    const ver = this.getVersionByCodename(input);
+    return this._distroInfo[ver];
+  }
+
+  /**
+   * Check if a given release has passed its EOL
+   * @param input A codename/semVer
+   * @returns false if still supported, true otherwise
+   */
+  public isEolLts(input: string): boolean {
+    const ver = this.getVersionByCodename(input);
+    const schedule = this.getSchedule(ver);
+    const endLts = schedule?.eol ?? null;
+    let end = schedule?.eol_lts ?? null;
+
+    // ubuntu: does not have eol_lts
+    // debian: only "Stable" has no eol_lts, old and oldold has both
+    if (!end) {
+      end = endLts;
+    }
+
+    if (end) {
+      const now = DateTime.now();
+      const eol = DateTime.fromISO(end);
+      return eol < now;
+    }
+
+    // istanbul ignore next
+    return true;
+  }
+
+  /**
+   * Get distro info for the release that has N other newer releases.
+   * Example: n=0 corresponds to the latest available release, n=1 the release before, etc.
+   * In Debian terms: N = 0 -> stable, N = 1 -> oldstable, N = 2 -> oldoldstalbe
+   * @param n
+   * @returns Distro info of the Nth latest release
+   */
+  public getNLatest(n: number): DistroInfoRecordWithVersion | null {
+    const len = this._sortedInfo.length - 1;
+    const i = len - n;
+
+    if (len >= i && i >= 0) {
+      return this._sortedInfo[i];
+    }
+
+    // istanbul ignore next
+    return null;
```

Comment:
I think we should throw an error for out of range ? Should never happen.

Otherwise we should create a dummy json file to test those cases. you could simply overwrite the data map for those test 😉

Reply:
removed the ignore comment and also covered floats arguments.

added unit tests for:
-  the out of bound
-  float